File: src/priv/bajdcc/util/lexer/automata/nfa/NFAEdgeData.java
package priv.bajdcc.util.lexer.automata.nfa;

import priv.bajdcc.util.lexer.automata.EdgeType;

/**
 * NFA边数据
 * 
 * @author bajdcc
 *
 */
public class NFAEdgeData {
	/**
	 * 边类型
	 */
	public EdgeType kAction = EdgeType.EPSILON;
	
	/**
	 * 数据
	 */
	public int param = -1;
}

File: src/main/java/com/bajdcc/util/lexer/automata/nfa/NFAEdgeData.kt
package com.bajdcc.util.lexer.automata.nfa

import com.bajdcc.util.lexer.automata.EdgeType

/**
 * NFA边数据
 * [type] 边类型
 * [param] 数据
 * @author bajdcc
 */
data class NFAEdgeData(var type: EdgeType = EdgeType.EPSILON,
                       var param: Int = -1)
